pkg/config/config_template.yaml

```diff
@@ -1,424 +1,781 @@
+## location. The procfs_path configuration parameter provides a mechanism to
+## override the standard default location: '/proc' - this setting trickles down to
+## integrations and affect their behavior if they rely on the psutil python package.
 #
-# logs_config:
+#  procfs_path: /proc
+
+## @param disable_py3_validation - boolean - optional - default: false
+## Disable Python3 validation of python checks.
 #
-#   Enable container log collection for all the containers (see ac_exclude to filter out containers)
-#   container_collect_all: false
+#  disable_py3_validation: false
+
+## @param secret_backend_command - string - optional
+## The Agent can call an external command to fetch secrets. It executes the command a maximum of
+## once per instance containing an encrypted password.
+## The Agent caches Secrets to avoid executing the secret_backend_command multiple times to fetch an
+## already known secret (useful when combined with Autodiscovery). This feature is still in beta.
+##
+## For more information see: https://github.com/DataDog/datadog-agent/blob/master/docs/agent/secrets.md
+##
+## secret_backend_command is the path to the script to execute. The script must belong to the same user used
+## to run the Agent. The Agent must have executable rights, and no rights for 'group' or 'other'.
+#
+#  secret_backend_command: <COMMAND_PATH>
+
+## @param secret_backend_arguments - list of strings - optional
+## If secret_backend_command is set, specify here a list of arguments to give to the command at each run.
 #
-#   Define the endpoint and port to hit when using a proxy for logs. The logs are forwarded in TCP
-#   therefore the proxy must be able to handle TCP connections.
-#   logs_dd_url: <endpoint>:<port>
+#  secret_backend_arguments:
+#    - <ARGUMENT_1>
+#    - <ARGUMENT_2>
+
+## @param secret_backend_output_max_size - integer - optional - default: 1024
+## The size in bytes of the buffer used to store the command answer (apply to both stdout and stderr)
 #
-#   Disable the SSL encryption (default to false). This parameter should only be used when logs are
-#   forwarded locally to a proxy. It is highly recommended to then handle the SSL encryption
-#   on the proxy side.
-#   logs_no_ssl: false
+#  secret_backend_output_max_size: 1024
+
+## @param secret_backend_timeout - integer - optional - default: 5
+## The timeout to execute the command in second
 #
-#   Global processing rules that are applied to all the logs. The available rules are
-#   "exclude_at_match", "include_at_match" and "mask_sequences". More information in the documentation:
-#   https://docs.datadoghq.com/logs/log_collection/?tab=tailexistingfiles#advanced-log-collection-functions
-#   processing_rules:
-#     - rule1_arg1
-#       rule1_arg2
-#     - rule2_arg1
-#       rule2_arg2
+#  secret_backend_timeout: 5
+
+{{ end -}}
+{{- if .LogsAgent }}
+
+#########################################
+## Logs Collection Agent Configuration ##
```

Comment:
Let's remove `Agent`. There is one agent.

Let's have `Logs Collection Configuration`